addons/arsenal/functions/fnc_onArsenalOpen.sqf

```diff
@@ -0,0 +1,293 @@
+                ((GVAR(virtualItems) select _index) select 1) pushBackUnique (_array select 1);
+            };
+
+            if !((_array select 2) isEqualTo "") then {
+                 ((GVAR(virtualItems) select _index) select 2) pushBackUnique (_array select 2);
+            };
+        };
+
+        // Accs for the weapons above
+        case 1: {
+            private _array = LIST_DEFAULTS select _index;
+
+            if !((_array select 0) isEqualTo []) then {
+                 {((GVAR(virtualItems) select _index) select 0) pushBackUnique _x} foreach (_array select 0);
+            };
+
+            if !((_array select 1) isEqualTo []) then {
+                {((GVAR(virtualItems) select _index) select 1) pushBackUnique _x} foreach (_array select 1);
+            };
+
+            if !((_array select 2) isEqualTo []) then {
+                 {((GVAR(virtualItems) select _index) select 2) pushBackUnique _x} foreach (_array select 2);
+            };
+
+            if !((_array select 3) isEqualTo []) then {
+                 {((GVAR(virtualItems) select _index) select 3) pushBackUnique _x} foreach (_array select 3);
+            };
+        };
+
+        // Inventory items
+        case 2: {
+                call FUNC(updateUniqueItemsList);
+            };
+
+        // The rest
+        default {
+            private _array = (LIST_DEFAULTS select _index) select {!(_x isEqualTo "")};
+            if !(_array isEqualTo []) then {
+                {(GVAR(virtualItems) select _index) pushBackUnique _x} foreach _array;
+            };
+        };
+    };
+};
+
+// Fill current items
+for "_index" from 0 to 15 do {
+    switch (_index) do {
+        case 0;
+        case 1;
+        case 2:{
+            GVAR(currentItems) set [_index, ((LIST_DEFAULTS select 0) select _index)];
+        };
+        case 3;
+        case 4;
+        case 5;
+        case 6;
+        case 7;
+        case 8;
+        case 9: {
+            GVAR(currentItems) set [_index, (LIST_DEFAULTS select _index) select 0];
+
+        };
+        case 10: {
+            {(GVAR(currentItems) select 15) pushBack _x} forEach (uniformItems GVAR(center));
+        };
+        case 11: {
+            {(GVAR(currentItems) select 16) pushBack _x} forEach (vestItems GVAR(center));
+        };
+        case 12: {
+            {(GVAR(currentItems) select 17) pushBack _x} forEach (backpackItems GVAR(center));
+        };
+        case 13: {
+            GVAR(currentItems) set [18, (primaryWeaponItems GVAR(center)) + (primaryWeaponMagazine GVAR(center))];
+        };
+        case 14: {
+            GVAR(currentItems) set [19, (secondaryWeaponItems GVAR(center)) + (secondaryWeaponMagazine GVAR(center))];
+        };
+        case 15: {
+            GVAR(currentItems) set [20, (handgunItems GVAR(center)) + (handgunMagazine GVAR(center))];
+        };
+    };
+};
+
+{
+    if (getText (configFile >> "CfgWeapons" >> _x >> "simulation") == "ItemMap") then {
```

Comment:
Again. See previous comment.
This is also appearing duplicated in quite a lot of places.. Might be a candidate for a standalone function.